File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.ts
function lengthOfLongestSubstring(s: string): number {

    var existed: Set<string> = new Set();
    var ans: number = existed.size;
    var stringLength: number = s.length;
    var startIndex: number = 0;
    for (let i = 0; i < stringLength; i++) {
        let element: string = s[i];
        while (existed.has(element)) {
            existed.delete(s[startIndex]);
            startIndex++;
        }
        existed.add(element);
        ans = existed.size > ans ? existed.size : ans
    }
    return ans;
};

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = ""
        ans_len = 0
        for i in s:
            while i in ans:
                ans = ans[1:]
            ans += i
            if len(ans) > ans_len:
                ans_len = len(ans)
        
        return ans_len
        